### PR TITLE
Past, Future and Min validation support

### PR DESCRIPTION
**Pull request by Ampretuzo**
Added some classes to validation module to support Past, Future and Min constraints. Added respective tests as well.

Note that since validation target type HAS to be non-parametrized or wildcard-parametrized it is necessary to have type checks in the code (It is not possible to have e.g. ```Property<Date>``` as target data type, we're stuck with plain ```Property```).

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/689)

### CODE REVIEW

**Comment by hendrikebbers** [comment]
@Ampretuzo since other PR are merged that add support for additional bean validation annotations you need to update your branch (merge master in it) before I can merge this one

**Comment by Ampretuzo** [comment]
Merged, I was too quick to notice conflicts :)

File: platform-extras/dolphin-platform-bean-validation/src/main/java/com/canoo/dp/impl/validation/FuturePropertyValidator.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Validator that adds Dolphin Platform property support for the {@link Future} annotation.
+ */
+public final class FuturePropertyValidator extends AbstractDateAndCalendarValidator<Future> {
+
+    @Override
+    protected boolean checkValidDate(Date date, ConstraintValidatorContext context) {
```
Comment (hendrikebbers) [inline platform-extras/dolphin-platform-bean-validation/src/main/java/com/canoo/dp/impl/validation/FuturePropertyValidator.java]: please add final to non modified fields & var
Add null check by using Assert.requireNotNull for all vars that will be accessed in a method

File: platform-extras/dolphin-platform-bean-validation/src/main/java/com/canoo/dp/impl/validation/AbstractDateAndCalendarValidator.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        super(Object.class);
+    }
+
+    protected abstract boolean checkValidDate(Date date, ConstraintValidatorContext context);
+
+    protected abstract boolean checkValidCalendar(Calendar calendar, ConstraintValidatorContext context);
+
+    @Override
+    protected boolean checkValid(Object value, ConstraintValidatorContext context) {
+        if (value instanceof Date) {
```
[the remark this answers is not in the document]
Comment (Ampretuzo) [reply platform-extras/dolphin-platform-bean-validation/src/main/java/com/canoo/dp/impl/validation/AbstractDateAndCalendarValidator.java]: AbstractPropertyValidator guarantees that checkValid method will be called with a non-null value. I'll add that to checkValid method javadoc.

File: platform-extras/dolphin-platform-bean-validation/src/main/java/com/canoo/dp/impl/validation/FuturePropertyValidator.java (hunk 1 above)
Comment (Ampretuzo) [reply platform-extras/dolphin-platform-bean-validation/src/main/java/com/canoo/dp/impl/validation/FuturePropertyValidator.java]: Same applies to null check as above, added javadocs to template methods.

**Comment by hendrikebbers** [comment]
THX :)

**Comment by Ampretuzo** [comment]
no problem :)
Currently I'm writing Max validation and abstract number validator like I did for Date-Calendar (All number validations will eventually implement that abstract class to support different data types).
Most importantly, there is a bug - null values might be accepted even for incompatible data types (null String is a valid Future at the moment).
I'll add that issue after Max validation PR.

**Pull request merged**